src/memline.c

```diff
@@ -1754,7 +1754,7 @@ ml_recover(int checkext)
 	for (idx = 1; idx <= lnum; ++idx)
 	{
 	    // Need to copy one line, fetching the other one may flush it.
-	    p = vim_strsave(ml_get(idx));
+	    p = vim_strnsave(ml_get(idx), ml_get_len(idx));
```

Comment:
is that an improvement? This will now do an `ml_get()` twice instead of once...

Reply:
I think it is. The first call will find the lnum line and save it's info in curbuf->b_ml, and the second will use it. This will avoid calling STRLEN() on a potentially long string in vim_strsave().